src/index.tsx

```diff
@@ -322,11 +327,23 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
 
     public render() {
         const {
+            children,
             trigger,
         } = this.props;
-
-        return React.cloneElement(trigger(), {
-            onClick: this.handleClick,
-        });
+        if (trigger) {
+            return React.cloneElement(trigger(), {
+                onClick: this.handleClick,
+            });
+        } else {
+            const value = {handlePrint: this.handleClick};
+            return <PrintContext.Provider value={value as IPrintContextProps}>
+                {children}
+            </PrintContext.Provider>
+        }
     }
 }
+
+export const useReactToPrint = (options: IReactToPrintProps) => {
```

Comment:
Similar to the above comment, is there a way to make this a no-op (just export `undefined` or something) if the user is on React 15 (which doesn't have `useMemo` and `useCallback`?